service/src/main/kotlin/xyz/lebkuchenfm/external/gemini/GeminiClient.kt

```diff
@@ -0,0 +1,139 @@
+package xyz.lebkuchenfm.external.gemini
+
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Ok
+import com.github.michaelbull.result.Result
+import dev.kord.rest.request.errorString
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.engine.okhttp.OkHttp
+import io.ktor.client.plugins.DefaultRequest
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.accept
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.URLProtocol
+import io.ktor.http.appendPathSegments
+import io.ktor.http.contentType
+import io.ktor.http.isSuccess
+import io.ktor.serialization.kotlinx.json.json
+import io.ktor.server.config.ApplicationConfig
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import xyz.lebkuchenfm.domain.radiopersonality.llmprompts.LlmPromptRaw
```

Comment:
I feel that we should convert from domain to client model or primitive types in GeminiRadioPersonality.kt

Reply:
Good idea 👍